exercises/practice/bowling/.meta/Example.roc

```diff
@@ -0,0 +1,131 @@
+module [Game, create, roll, score]
+
+Frame : [
+    Ball1 U64, # unfinished frame
+    Ball2 U64 U64,
+    Spare U64 U64,
+    Strike,
+    SpareFill U64,
+    StrikeFill1 U64, # unfinished frame
+    StrikeFill2 U64 U64,
+]
+
+Game := { frames : List Frame }
+
+create : { previousRolls ? List U64 } -> Result Game [MoreThan10Pins, GameOver]
+create = \{ previousRolls ? [] } ->
+    newGame = @Game { frames: [] }
+    when previousRolls is
+        [] -> Ok newGame
+        rolls ->
+            rolls
+            |> List.walkUntil (Ok newGame) \state, pins ->
+                when state is
+                    Ok game ->
+                        when game |> roll pins is
+                            Ok updatedGame -> Continue (Ok updatedGame)
+                            Err err -> Break (Err err)
+
+                    Err _ -> crash "Impossible: state can never be an error since we never start or continue with an error"
```

Comment:
```suggestion
create = \{ previousRolls ? [] } ->
    List.walkTry previousRolls (@Game { frames: [] }) \game, pins ->
        roll game pins
```
This function could be simplified like this.

Reply:
Aaaaahhhh, awesome, `List.walkTry` works so much better! Thanks!